lib/pbench/server/api/resources/query_apis/metadata_index/table_of_content.py

```diff
@@ -0,0 +1,95 @@
+from logging import Logger
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    JSON,
+    Schema,
+    Parameter,
+    ParamType,
+)
+from pbench.server.api.resources.query_apis import CONTEXT
+from pbench.server.api.resources.query_apis.metadata_index import RunIdBase
+
+
+class TableOfContent(RunIdBase):
+    """
+    Get the names of controllers within a date range.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("run_id", ParamType.STRING, required=True),
+                Parameter("parent", ParamType.STRING, required=True),
+            ),
+        )
+
+    def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
+        parent = json_data.get("parent")
+        run_id = context["run_id"]
+        dataset = context["dataset"]
+
+        self.logger.info(
+            "Discover Table of Content for run_id {} and directory {}", run_id, parent,
+        )
+
+        # Retrieve the ES indices that belong to this run_id from the metadata
+        # table
+
+        indices = self.get_index(dataset, "run-toc")
+        if not indices:
+            self.logger.debug(
+                f"Found no indices matching the prefix run-toc"
+                f"for a dataset {dataset!r}"
+            )
+            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+
+        return {
+            "path": f"/{indices}/_search",
+            "kwargs": {
+                "json": {
+                    "size": 100,
```

Comment:
Also why the number 100, can we keep it default i.e. 10000?

Reply:
I don't know how likely it is that we'll find a tarball directory with more than 100 subdirectories; but if it's realistic to have such a thing, we would need to support a scrolling API to get through them all. (Although, in this case, we could sort by path and use "search after" instead of a scroll.) I suspect we're much less likely to have to worry about this with a 10000 document limit, so I would strongly support Nikhil's recommended change.

Reply:
earlier when we were querying the elasticsearch there were 10 consecutive subdirectories and we were not able to fetch them all at once. which lead to adding this size attribute here.

Reply:
After adding size we were able to see a subdirectory with no. 7 in it, which we were not able to earlier which led to adding this. I will confirm again if I am able to see it without this.

Reply:
I think the default is `10`, and that 10,000 is the _maximum_...so, I do think we need to specify a size (as I mentioned below), but it would be good as I suggested above to have a symbolic constant for it.